src/python/module/smartredis/dataset_utils.py

```diff
@@ -0,0 +1,141 @@
+# BSD 2-Clause License
+#
+# Copyright (c) 2021-2022, Hewlett Packard Enterprise
+# All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions are met:
+#
+# 1. Redistributions of source code must retain the above copyright notice, this
+#    list of conditions and the following disclaimer.
+#
+# 2. Redistributions in binary form must reproduce the above copyright notice,
+#    this list of conditions and the following disclaimer in the documentation
+#    and/or other materials provided with the distribution.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+# AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+# DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+# FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+# DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+# SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+# CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+# OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+from .dataset import Dataset
+from .util import Dtypes, exception_handler, typecheck
+import xarray as xr
+
+from itertools import permutations
+from .error import *
+
+
+class DatasetConverter:   
+
+    @staticmethod
+    def add_metadata_for_xarray(dataset, data_names, dim_names, coord_names=None, attr_names=None):
+        """Extract metadata from a SmartRedis dataset and add it to 
+            dataformat specific fieldnames 
+
+            :param dataset: a Dataset instance
+            :type dataset: Dataset
+            :param data_names: variable data field name
+            :type data_names: list[str] or str 
+            :param dim_names: dimension field names.
+            :type dim_names: list[str]
+            :param coord_names: coordinate field names. Defaults to None.
+            :type coord_names: list[str], optional
+            :param attr_names: attribute field names. Defaults to None.
+            :type attr_names: list[str], optional
+        """        
+        
+        typecheck(dataset, "dataset", Dataset)
+        typecheck(data_names, "data_names", list)
+        for data_name in data_names:
+            typecheck(data_name, "data_name", str)
+        typecheck(dim_names, "dim_names", list)
+        for dim_name in dim_names:
+            typecheck(dim_name, "dim_name", str)
```

Comment:
Rather than repeating these loops, please add a typecheck_list() utility that takes a list arg and a list element type and performs this check.